content/docs/installation/continuous-deployment-and-gitops.md

```diff
@@ -112,3 +112,142 @@ Check the cert-manager logs for warnings and errors:
 ```bash
 kubectl logs -n cert-manager -l app.kubernetes.io/instance=cert-manager --prefix --all-containers
 ```
+
+
+## Using ArgoCD
+Argo CD is a declarative, GitOps continuous delivery tool for Kubernetes.
+
+### Pre-requisites
+Ensure the following are in place before proceeding:
+- A Kubernetes cluster
+- ArgoCD deployed on the Kubernetes cluster: [installation guide](https://argo-cd.readthedocs.io/en/stable/getting_started/)
+- Optional: A GitOps repository connected with ArgoCD: [setup guide](https://argo-cd.readthedocs.io/en/stable/user-guide/private-repositories/)
+
+### Setting up cert-manager
+1. Create an [ArgoCD Application](https://argo-cd.readthedocs.io/en/stable/operator-manual/declarative-setup/#applications) manifest file with the provided configuration to set up cert-manager.
+
+    ```yaml
+    # application.yaml
+    apiVersion: argoproj.io/v1alpha1
+    kind: Application
+    metadata:
+      name: cert-manager
+      namespace: argocd
+      annotations:
+        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
+      finalizers:
+        - resources-finalizer.argocd.argoproj.io
+    spec:
+      destination:
+        namespace: cert-manager
+        server: https://kubernetes.default.svc
+      project: default
+      source:
+        chart: cert-manager
+        repoURL: https://charts.jetstack.io
+        targetRevision: [[VAR::cert_manager_latest_version]]
+        helm:
+          values: |
+            installCRDs: true
+      syncPolicy:
+        automated:
+          prune: true
+          selfHeal: true
+        syncOptions:
+          - CreateNamespace=true
```

Comment:
Normally by using ServerSideApply and ServerSide diffing, there should be no issues with mutating webhooks:
```suggestion
    ```yaml
    # application.yaml
    apiVersion: argoproj.io/v1alpha1
    kind: Application
    metadata:
      name: cert-manager
      namespace: argocd
      finalizers:
        - resources-finalizer.argocd.argoproj.io
      annotations:
        argocd.argoproj.io/compare-options: ServerSideDiff=true
    spec:
      destination:
        namespace: cert-manager
        server: https://kubernetes.default.svc
      project: default
      source:
        chart: cert-manager
        repoURL: https://charts.jetstack.io
        targetRevision: [[VAR::cert_manager_latest_version]]
        helm:
          values: |
            installCRDs: true
      syncPolicy:
        automated:
          prune: true
          selfHeal: true
        syncOptions:
          - CreateNamespace=true
          - ServerSideApply=true
```

NOTE: I still have to verify this.
This change is based on https://argo-cd.readthedocs.io/en/stable/user-guide/diff-strategies/ and https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/#server-side-apply.